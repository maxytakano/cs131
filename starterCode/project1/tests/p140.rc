structdef MYS {
    int x;
    int y;

};



function: int foostruct( MYS homer){//func_1
    return 0;
}

function: void foostruct( MYS *marge){//func_2
}
function: void foostruct( MYS &marge, MYS homer){//func_3
}
function: int foostruct( MYS bart, MYS &lisa){//error func_4.
    return 0;
}

function: int barstruct( MYS *bart, MYS &lisa){ //func_5
    return 0;
}



function: int main(){
    MYS x;
    int y;


    foostruct( x);//func_1
    foostruct( y);//error

    foostruct( x, x);//func_3
    foostruct(&x);//func_2

    barstruct( &x, x);//func_5    
    barstruct( x, x);//error
    barstruct( &x, &x);//error
    barstruct( &x, x, &x);//error
    barstruct( &x, x, y);//error

    return 0;
}
