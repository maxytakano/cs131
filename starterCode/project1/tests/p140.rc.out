Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 17: 
  Duplicate declaration of overloaded function foostruct.
Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 33: 
  Illegal call to overloaded function foostruct.
Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 39: 
  Argument of type MYS not assignable to value parameter bart, of type MYS*.
Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 40: 
  Argument of type MYS* not equivalent to reference parameter lisa, of type MYS.
Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 41: 
  Number of arguments (3) differs from number of parameters (2).
Error, "/tmp/cs131_testrunner_bATaux/file.rc", line 42: 
  Number of arguments (3) differs from number of parameters (2).
Compile: failure.
